src/app/home/home.component.ts

```diff
@@ -1,27 +1,54 @@
 import { Component, OnInit } from '@angular/core';
-import { HomeService } from './home.service';
+import { HomeService, TodoItem } from './home.service';
 import { Router } from '@angular/router';
 import { Activity } from '../project/project.service';
+import { UtilsService } from '@services/utils.service';
 
 @Component({
   selector: 'app-home',
   templateUrl: 'home.component.html',
   styleUrls: ['home.component.scss']
 })
 export class HomeComponent implements OnInit {
-  progress:number = 80;
-  programName:string;
-  todoItems = [];
-  activity :Activity;
+  progress: number = 0;
+  loadingProgress: boolean = true;
+  programName: string;
+  todoItems: Array<TodoItem> = [];
+  loadingTodoItems: boolean = true;
+  activity: Activity;
+  loadingActivity: boolean = true;
   
   constructor (
     private router: Router,
-    private homeService: HomeService 
+    private homeService: HomeService,
+    private utils: UtilsService
   ) {}
 
   ngOnInit() {
-    this.todoItems = this.homeService.getTodoItems();
-    this.activity = this.homeService.getCurrentActivity();
+    this.homeService.getTodoItems()
+      .subscribe(todoItems => {
```

Comment:
@shawnm0705 getTodoItems() in develop branch is no longer return observable, could you please check?
https://github.com/intersective/practera-app-v2/blob/develop/src/app/home/home.service.ts#L49-L55